### PR TITLE
[PLA-1340] Support PermittedExtrinsics fuel tank rule.

### PR DESCRIPTION
**Pull request by v16Studios**

### CODE REVIEW

File: src/GraphQL/Schemas/Primary/Substrate/Mutations/BatchTransferMutation.php [hunk 1]
```diff
@@ -112,61 +113,52 @@ function ($recipient) use ($blockchainService, $walletService) {
...
-                [
-                    'method' => $this->getMutationName(),
-                    'encoded_data' => $this->resolveBatch($args['collectionId'], $recipients, false, $serializationService),
-                    'idempotency_key' => $args['idempotencyKey'] ?? Str::uuid()->toString(),
-                    'deposit' => $this->getDeposit($args),
-                    'simulate' => $args['simulate'],
-                ],
-                signingWallet: $this->getSigningAccount($args)
-            ),
+            $this->storeTransaction($args, $encodedData),
```
Comment (enjinabner) [inline src/GraphQL/Schemas/Primary/Substrate/Mutations/BatchTransferMutation.php:125]: Does this affect the beam part as well https://github.com/enjin/platform-beam/blob/master/src/Commands/BatchProcess.php#L211?

Comment (enjinabner) [reply src/GraphQL/Schemas/Primary/Substrate/Mutations/BatchTransferMutation.php:125]: Also what about platform-fuel-tanks library, does it need update?

Comment (v16Studios) [reply src/GraphQL/Schemas/Primary/Substrate/Mutations/BatchTransferMutation.php:125]: > Does this affect the beam part as well https://github.com/enjin/platform-beam/blob/master/src/Commands/BatchProcess.php#L211?

I think we can leave the beam implementation as-is for now, the only meaningful change would be to use the new TransactionSerializer facade where it encodes that data.  The`StoresTransactions` trait is specifically for the transaction mutations at the moment as it uses the `getMutationName()` function available on the mutation, but not in the Beam BatchProcess command.

Comment (enjinabner) [reply src/GraphQL/Schemas/Primary/Substrate/Mutations/BatchTransferMutation.php:125]: lgtm but I'm a bit hesitant to have this on the next release, not enough time for testing.

**Review by enjinabner — APPROVED** [review]

**Review by leonardocustodio — APPROVED** [review]

**Pull request merged**